src/Core/ServiceBuilder.php

```diff
@@ -317,12 +342,15 @@ public function trace(array $config = [])
      * ```
      *
      * @param array $config [optional] Configuration options. See
-     *        {@see Google\Cloud\ServiceBuilder::__construct()} for the available options.
+     *        {@see Google\Cloud\Core\ServiceBuilder::__construct()} for the available options.
      * @return VisionClient
      */
     public function vision(array $config = [])
     {
-        return new VisionClient($config ? $this->resolveConfig($config) : $this->config);
+        if (class_exists('Google\Cloud\Vision\VisionClient')) {
```

Comment:
All of these should use the `::class` magic constant. i.e. `VisionClient::class`.

Additionally, it seems almost all of this method is duplicated, with the exception of the class and package name. Perhaps we could abstract it out further.

```php
public function vision(array $config = [])
{
    return $this->createClient(Vision::class, 'vision');
}
```